### PR TITLE
Update dependencies (nightly-2021-10-15)

### PR DESCRIPTION
**Pull request by viper-admin**
* [x] Update rustc version to `nightly-2021-10-15`.
* [ ] Manualy update outdated dependencies (see the list below).
* [x] Manualy run `cargo update`.

<details><summary>List of direct outdated dependencies:</summary>

```
$ cargo outdated --root-deps-only --workspace

vir
================
Name         Project  Compat  Latest  Kind    Platform
----         -------  ------  ------  ----    --------
index_vec    0.1.2    ---     0.1.3   Normal  ---
proc-macro2  1.0.29   ---     1.0.30  Normal  ---
quote        1.0.9    ---     1.0.10  Normal  ---
syn          1.0.76   ---     1.0.80  Normal  ---
thiserror    1.0.29   ---     1.0.30  Normal  ---

vir-gen
================
Name         Project  Compat  Latest  Kind    Platform
----         -------  ------  ------  ----    --------
proc-macro2  1.0.29   ---     1.0.30  Normal  ---
quote        1.0.9    ---     1.0.10  Normal  ---
syn          1.0.76   ---     1.0.80  Normal  ---

prusti-contracts
================
Name      Project  Compat  Latest  Kind         Platform
----      -------  ------  ------  ----         --------
trybuild  1.0.45   ---     1.0.50  Development  ---

prusti-contracts-impl
================
Name         Project  Compat  Latest  Kind    Platform
----         -------  ------  ------  ----    --------
proc-macro2  1.0.29   ---     1.0.30  Normal  ---
quote        1.0.9    ---     1.0.10  Normal  ---

prusti-specs
================
Name         Project  Compat  Latest  Kind    Platform
----         -------  ------  ------  ----    --------
proc-macro2  1.0.29   ---     1.0.30  Normal  ---
quote        1.0.9    ---     1.0.10  Normal  ---
syn          1.0.76   ---     1.0.80  Normal  ---

prusti-contracts-internal
================
Name         Project  Compat  Latest  Kind    Platform
----         -------  ------  ------  ----    --------
proc-macro2  1.0.29   ---     1.0.30  Normal  ---

prusti-server
================
Name     Project  Compat  Latest  Kind    Platform
----     -------  ------  ------  ----    --------
futures  0.1.31   ---     0.3.17  Normal  ---
reqwest  0.9.24   ---     0.11.5  Normal  ---
tokio    0.1.22   ---     1.12.0  Normal  ---
warp     0.1.23   ---     0.3.1   Normal  ---

prusti-launch
================
Name   Project  Compat  Latest  Kind    Platform
----   -------  ------  ------  ----    --------
ctrlc  3.2.0    ---     3.2.1   Normal  ---
nix    0.22.1   ---     0.23.0  Normal  cfg(unix)
```
</details>

@fpoli could you take care of this?

### CODE REVIEW

**Comment by fpoli** [comment]
The bot force-pushed to this branch. What's the status of the update?

**Comment by Aurel300** [comment]
The update was not trivial because of the introduction of `ShallowInitBox`, then I was focusing on the paper submission… We discussed a possible (hacky) fix with @vakaras to ignore the `u8` raw pointer if it is eventually used in the `ShallowInitBox` operation. Vytautas was also saying parts of this encoding will be refactored anyway.

**Comment by fpoli** [comment]
Oh, I thought that raw pointers were already encoded as opaque types in Prusti. From the MIR that I looked at the raw pointer used by `ShallowInitBox` directly comes from an `alloc` function call and it doesn't carry any information. I agree that we can ignore it.

**Comment by vakaras** [comment]
@Aurel300 @fpoli Is anyone going to take care of this one? We must not slip too much behind.

**Comment by fpoli** [comment]
@Aurel300 Could you merge and push your changes, even if incomplete? The bot force-pushed them away.

File: prusti-viper/src/encoder/procedure_encoder.rs [hunk 1]
```diff
@@ -5322,31 +5322,75 @@ impl<'p, 'v: 'p, 'tcx: 'v> ProcedureEncoder<'p, 'v, 'tcx> {
...
+            location,
+            vir::AssignKind::Move,
+            false
+        )?;
+
+        // Allocate `box_content`
+        // TODO: This is also encoding initialization, which should be avoided because both
+        //   `Rvalue::NullaryOp(NullOp::Box)` and `Rvalue::ShallowInitBox` don't initialize the
+        //   content of the box.
+        stmts.extend(self.encode_havoc_and_initialization(&box_content));
```
Comment (fpoli) [inline prusti-viper/src/encoder/procedure_encoder.rs:5390]: We have to plan how to solve this. If we don't inhale the permissions here, then every assignment to a box should assume that the box is potentially uninitialized (or check if it has already been initialized), which would be less efficient for the verifier.

**Comment by fpoli** [comment]
`cargo outdated` crashes. I'll merge this in the meantime.

`cargo-outdated` issues: https://github.com/kbknapp/cargo-outdated/issues/286 and https://github.com/kbknapp/cargo-outdated/issues/287.

**Pull request merged**